framework/src/meshgenerators/Poly2TriMeshGenerator.C

```diff
@@ -0,0 +1,378 @@
+
+#include "libmesh/elem.h"
+#include "libmesh/int_range.h"
+#include "libmesh/mesh_modification.h"
+#include "libmesh/mesh_serializer.h"
+#include "libmesh/mesh_triangle_holes.h"
+#include "libmesh/parsed_function.h"
+#include "libmesh/poly2tri_triangulator.h"
+#include "libmesh/unstructured_mesh.h"
+
+registerMooseObject("MooseApp", Poly2TriMeshGenerator);
+
+InputParameters
+Poly2TriMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  MooseEnum algorithm("BINARY EXHAUSTIVE", "BINARY");
+
+  params.addRequiredParam<MeshGeneratorName>(
+      "boundary", "The input MeshGenerator that defines the output mesh outer boundary.");
+  params.addParam<std::vector<BoundaryName>>(
+      "input_boundary_names", "2D-input-mesh boundaries defining the output mesh outer boundary");
+  params.addParam<std::vector<SubdomainName>>(
+      "input_subdomain_names", "1D-input-mesh subdomains defining the output mesh outer boundary");
+  params.addParam<unsigned int>("add_nodes_per_boundary_segment",
+                                0,
+                                "How many more nodes to add in each outer boundary segment.");
+  params.addParam<bool>(
+      "refine_boundary", true, "Whether to allow automatically refining the outer boundary.");
+
+  params.addParam<subdomain_id_type>(
+      "output_subdomain_id", 0, "Subdomain id to set on new triangles.");
+  params.addParam<SubdomainName>("output_subdomain_name",
+                                 "Subdomain name to set on new triangles.");
+
+  params.addParam<boundary_id_type>(
+      "output_boundary_id", 0, "Boundary id to set on new outer boundary.");
+  params.addParam<std::vector<boundary_id_type>>(
+      "hole_boundary_ids", "Boundary ids to set on holes.  Numbered up from 1 by default.");
+
+  params.addParam<bool>(
+      "verify_holes",
+      true,
+      "Verify holes do not intersect boundary or each other.  Asymptotically costly.");
+
+  params.addParam<bool>("smooth_triangulation",
+                        false,
+                        "Whether to do Laplacian mesh smoothing on the generated triangles.");
+  params.addParam<std::vector<MeshGeneratorName>>(
+      "holes", std::vector<MeshGeneratorName>(), "The MeshGenerators that define mesh holes.");
+  params.addParam<std::vector<bool>>(
+      "stitch_holes", std::vector<bool>(), "Whether to stitch to the mesh defining each hole.");
+  params.addParam<std::vector<bool>>("refine_holes",
+                                     std::vector<bool>(),
+                                     "Whether to allow automatically refining each hole boundary.");
+  params.addRangeCheckedParam<Real>("desired_area",
+                                    std::numeric_limits<Real>::max(),
+                                    "desired_area>0",
+                                    "Desired (maximum) triangle area.");
+  params.addParam<std::string>("desired_area_func",
+                               std::string(),
```

Comment:
Is the default actually required here? I don't think it is

Reply:
If I take it out, every use case fails with
`The parameter "desired_area_func" is being retrieved before being set`

And to avoid that?  This seemed like a better idiom than adding a ternary operator and an isParamValid test to constructor initializer list entries.

Reply:
Yep. Fine by me. was just curious